### PR TITLE
Use consistent naming for opsportal roles

### PR DESCRIPTION
**Pull request by jr0d**
* Change naming convention to `opsportal-<component>-view|edit|admin`

### CODE REVIEW

**Comment by alejandroEsc** [comment]
> * Change naming convention to `opsportal-<component>-view|edit|admin`

can we get an idea about the motivation for this change?

File: stable/opsportal/templates/ingress-opsportal-roles.yaml [hunk 1]
```diff
@@ -32,7 +32,6 @@ rules:
 - nonResourceURLs:
   - {{ .Values.opsportalRBAC.path | trimSuffix "/"}}
   - {{ .Values.opsportalRBAC.path | trimSuffix "/" }}/*
-  - {{ .Values.opsportalRBAC.graphqlPath | trimSuffix "/"}}
```
Comment (dkoshkin) [inline stable/opsportal/templates/ingress-opsportal-roles.yaml]: Is this intentional?

Comment (dkoshkin) [inline stable/opsportal/templates/ingress-opsportal-roles.yaml]: Don't want the original message to get lost.

Comment (jr0d) [reply stable/opsportal/templates/ingress-opsportal-roles.yaml]: Yes, it was redundant

**Comment by jieyu** [comment]
@joejulian the role name is a user facing thing (i.e., if user want to grant some user to opsportal, they'll have to use that name).

By default, the release name will include `-kubeaddons` suffix (e.g., `opsportal-kubeaddons-admin`). The `kubeaddons` part is really unfortunate and should not be known by the user.

I think for all the opsportal related roles, we should use consistent naming scheme like the following so that it's intuitive to the user.
`opsportal-<COMPONENT>-admin|edit|view`

cc @chrisgaun @samba @GoelDeepak

**Comment by joejulian** [comment]
Is there any chance that someone would ever want this installed twice? Different departments, or projects, or something that might want to manage their own permission sets?

**Comment by jieyu** [comment]
@joejulian maybe. We could in the future add a configurable option to add a prefix to those role names to solve that use case. So far, we don't have that use case.

**Comment by joejulian** [comment]
https://v2.helm.sh/docs/chart_best_practices/#resource-naming-in-templates

Please update the _PR description_ to explain why we're not following best practices then I'll change my review.

**Comment by joejulian** [comment]
How did this get merged without the change request being addressed?

**Comment by GoelDeepak** [comment]
wouldn't this naming scheme fail during the upgrade? It will conflict with the existing object

**Comment by jieyu** [comment]
@GoelDeepak fail during upgrade from 1.3.0 to 1.3.1?

**Comment by joejulian** [comment]
Fail during upgrade from 0.2.2 to 0.2.3.

**Comment by GoelDeepak** [comment]
I was referring to future failures. Say we release it and then in future if we are trying to upgrade this chart. Would there be an issue?

**Comment by jieyu** [comment]
@joejulian 0.22 is not in any release.

@GoelDeepak no, it won't be an issue.

**Comment by GoelDeepak** [comment]
> @GoelDeepak no, it won't be an issue.

Then I am good 👍

**Comment by joejulian** [comment]
0.2.2 isn't released as an addon, sure, but the question remains because the 0.1.34 chart was and it had different names for the objects. The same upgrade question still applies.

**Comment by jieyu** [comment]
the cluster roles are newly introduced in 0.2.x. So upgrading from 0.1.34 shouldn't be a problem

**Comment by GoelDeepak** [comment]
@joejulian might be confused with the diff. @joejulian none of these changes are released yet. The changes that you see in this PR is from one unreleased version to another unreleased version

**Pull request merged**